sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -1392,30 +1419,47 @@ public DigitalTwinsServiceVersion getServiceVersion() {
      *
      * The result is then consumed by one or many destination endpoints (subscribers) defined under {@link EventRoute}
      * These event routes need to be set before publishing a telemetry message, in order for the telemetry message to be consumed.
      * @param digitalTwinId The Id of the digital twin.
+     * @param messageId A unique message identifier (within the scope of the digital twin id) that is commonly used for de-duplicating messages. Defaults to a random UUID if argument is null.
      * @param payload The application/json telemetry payload to be sent. payload can be a raw json string or a strongly typed object like a Dictionary.
-     * @param publishTelemetryRequestOptions The additional information to be used when processing a telemetry request.
+     * @param options The optional parameters for this request. If null, the default option values will be used.
      * @return A {@link Response} containing an empty mono.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Void>> publishTelemetryWithResponse(String digitalTwinId, Object payload, PublishTelemetryRequestOptions publishTelemetryRequestOptions) {
-        return withContext(context -> publishTelemetryWithResponse(digitalTwinId, payload, publishTelemetryRequestOptions, context));
+    public Mono<Response<Void>> publishTelemetryWithResponse(String digitalTwinId, String messageId, Object payload, DigitalTwinsSendTelemetryOptions options) {
+        return withContext(context -> publishTelemetryWithResponse(digitalTwinId, messageId, payload, options, context));
     }
 
-    Mono<Response<Void>> publishTelemetryWithResponse(String digitalTwinId, Object payload, PublishTelemetryRequestOptions publishTelemetryRequestOptions, Context context) {
+    Mono<Response<Void>> publishTelemetryWithResponse(String digitalTwinId, String messageId, Object payload, DigitalTwinsSendTelemetryOptions options, Context context) {
+        if (messageId == null || messageId.isEmpty()) {
+            messageId = UUID.randomUUID().toString();
+        }
+
+        com.azure.digitaltwins.core.implementation.models.DigitalTwinsSendTelemetryOptions protocolLayerOptions;
+        if (options == null) {
+            options = new DigitalTwinsSendTelemetryOptions();
+            protocolLayerOptions = null;
+        }
+        else {
+            protocolLayerOptions = new com.azure.digitaltwins.core.implementation.models.DigitalTwinsSendTelemetryOptions()
+                .setTraceparent(options.getTraceparent())
+                .setTracestate(options.getTracestate());
+        }
+
         return protocolLayer.getDigitalTwins().sendTelemetryWithResponseAsync(
             digitalTwinId,
-            publishTelemetryRequestOptions.getMessageId(),
+            messageId,
             payload,
-            publishTelemetryRequestOptions.getTimestamp().toString(),
+            options.getTimestamp().toString(),
```

Comment:
q - what is the default value of `OffsetDateTime`?

Reply:
this feels like another parameter that we should send a default value (current timestamp), if the user doesn't supply a value

Reply:
```java
OffsetDateTime.now(ZoneOffset.UTC);
```

Reply:
It's possible that we could make the default value null, and just not send that header though

Reply:
Oh, I actually prefer sending the current time as default value. Why do you feel sending null is better?

Reply:
I don't. The current implementation sends the timestamp by default. The only value I'd see in not sending the timestamp by default is to save on bytes over the wire

Reply:
Sure, but for this case, I'd say that the timestamp information is valuable. My vote is to send the current timestamp, if user doesn't provide it.

Reply:
Then I'll keep the implementation as is. If the user passes in a null options object, we will send the current timestamp for them